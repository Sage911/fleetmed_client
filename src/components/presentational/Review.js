import React from 'react'

const Review = (props) => {

  let id = props.id
  let feedback = props.feedback

  return (
    <li>
      {id + ':   ' + feedback}
      <button onClick={props.delete}>Delete</button>
    </li>
  )

}

export default Review
