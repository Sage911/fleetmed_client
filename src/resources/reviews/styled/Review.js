import styled from 'styled-components'
import Review from '../Review'

const styledReview = styled(Review)`
  color: pink;
  font-family: 'Open Sans'
`

export default styledReview
