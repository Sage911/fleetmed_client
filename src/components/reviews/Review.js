import React from 'react'
import styled from 'styled-components'
// styled-components allows us to add named styles to html tags as well as personal and 3rd party components

const Review = (props) => {

  return (
    <li className={props.className}>
      {props.id + ':   ' + props.feedback}
      <button onClick={props.delete}>Delete</button>
    </li>
  )

}

const styledReview = styled(Review)`
  color: pink;
  font-family: 'Open Sans'
`

export default styledReview
