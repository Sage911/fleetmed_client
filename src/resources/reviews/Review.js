import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Review = props => (
  <li className={props.className}>
    {`${props.id}: ${props.feedback}`}
    <button onClick={props.delete}>Delete</button>
  </li>
)

Review.propTypes = {
  className: PropTypes.string,
  id: PropTypes.number.isRequired,
  feedback: PropTypes.string.isRequired,
  delete: PropTypes.func.isRequired
}

Review.defaultProps = {
  className: null
}

const styledReview = styled(Review)`
  color: pink;
  font-family: 'Open Sans'
`

export default Review
