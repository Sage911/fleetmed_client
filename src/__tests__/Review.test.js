import React from 'react'
import { mount } from 'enzyme'
import Review from 'resources/reviews/Review'

describe('<Review />', () => {
  let review

  beforeEach(() => {
    const props = {
      id: 1,
      feedback: 'Food was good',
      delete: () => {}
    }

    review = mount(
      <Review {...props} />
    )
  })

  afterEach(() => {
    review.unmount()
  })

  it('renders', () => {
    expect(review.text()).toMatchSnapshot()
  })
})
