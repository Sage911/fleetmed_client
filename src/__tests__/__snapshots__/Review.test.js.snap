// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> renders 1`] = `"1: Food was goodDelete"`;
